Sources/SwiftGenKit/Parsers/Strings/StringsDictEntry.swift

```diff
@@ -0,0 +1,210 @@
+      throw Strings.ParserError.invalidFormat(
+        reason: """
+        Entry "\(entry)" expects either "\(CodingKeys.formatKey.stringValue)" or \
+        "\(CodingKeys.variableWidthRules.stringValue)" but got either neither or both
+        """
+      )
+    case let (.some(formatKey), .none):
+      let variableNames = StringsDict.variableNamesFromFormatKey(formatKey)?.map { $0.name } ?? []
+      let variables = try variableNames.reduce(into: [PluralEntry.Variable]()) { variables, variableName in
+        let variableRule = try container.decode(PluralEntry.VariableRule.self, forKey: CodingKeys(key: variableName))
+        // Combination of the format specifiers of the `PlaceholderType.float` and `PlaceholderType.int`.
+        guard ["a", "e", "f", "g", "d", "i", "o", "u", "x"].contains(variableRule.valueTypeKey.suffix(1)) else {
+          throw Strings.ParserError.invalidVariableRuleValueType(
+            variableName: variableName,
+            valueType: variableRule.valueTypeKey
+          )
+        }
+        variables.append(PluralEntry.Variable(name: variableName, rule: variableRule))
+      }
+      self = .pluralEntry(PluralEntry(formatKey: formatKey, variables: variables))
+    case let (.none, .some(variableWidthRules)):
+      self = .variableWidthEntry(VariableWidthEntry(rules: variableWidthRules))
+    }
+  }
+}
+
+// - MARK: Helpers
+
+extension StringsDict {
+  typealias VariableNameResult = (name: String, range: Range<String.Index>, positionalArgument: Int?)
+
+  /// Parses variable names and their ranges from a `NSStringLocalizedFormatKey`.
+  /// Every variable name that is contained within the format key is preceded
+  /// by the %#@ characters or by the positional variants, e.g. %1$#@, and followed by the @ character.
+  ///
+  /// - Parameter formatKey: The formatKey from which the variable names should be parsed.
+  /// - Returns: An array of discovered variable names, their range within the `formatKey` and the positional argument.
+  // swiftlint:disable:next discouraged_optional_collection
+  private static func variableNamesFromFormatKey(_ formatKey: String) -> [VariableNameResult]? {
+    let pattern = #"(%(?>(\d)\$)?#@([\w\.\p{Pd}]+)@)"#
+    guard let regex = try? NSRegularExpression(pattern: pattern, options: []) else { return nil }
+    let nsrange = NSRange(formatKey.startIndex..<formatKey.endIndex, in: formatKey)
+    let matches = regex.matches(in: formatKey, options: [], range: nsrange)
+
+    return matches.compactMap { match -> VariableNameResult? in
+      // 1st capture group is the whole format string including delimeters
+      let fullMatchNSRange = match.range(at: 1)
+      // 2nd capture group is the positional argument of the format string (Optional!)
+      let positionalArgumentNSRange = match.range(at: 2)
+      // 3rd capture group is the name of the variable, the string in between the delimeters
+      let nameNSRange = match.range(at: 3)
```

Comment:
Just took a closer look at your regex and noticed some stuff:

1. No need to wrap the whole regex into a `(…)` capture group, as `match.range(at: 0)` can already return the info for the whole match. So we can remove those and decrement the `match.range(at: N)` values by one in the following lines. As a bonus, would also be nice to document the less-common `(?>…)` and `\p{Pd}` matchers

```diff
+    // `(?>…)` = non-capturing atomic group - https://www.regular-expressions.info/atomic.html
+    // `\p{Pd}` = "Dash Punctuation" - https://www.regular-expressions.info/unicode.html#category
-    let pattern = #"(%(?>(\d)\$)?#@([\w\.\p{Pd}]+)@)"#
+    let pattern = #"%(?>(\d)\$)?#@([\w\.\p{Pd}]+)@"#
```

2. Also, the regex would fail to match any positional character >9 (i.e. more than one digit), while `Foundation` supports those (just double-checked). Never seen a project that used more than 9 parameters in the real world, but you never know…

We might want to write a unit test for that first btw (e.g. modify an existing `.stringsdict` fixture to put an entry with more than 9 placeholders in it), ensure the test is red, then fix the regex to make it green.

Note that in `PlaceholderTypes.swift` we use `[1-9]\\d*` (and not just `\d+`), to ensure that the number doesn't start with `0`. In practice according to my tests a leading zero is valid… as long as the whole position isn't the number zero:
    - `String(format: "%0$@ - %02$@", "one","two")` will return `0@ - one`
    - `String(format: "%0$@ - %02$@ - %1$@", "one","two")` will generate `"0@ - two - one"`

So the `%0$@` is not considered a placeholder and generates a `0@` literal instead, while `%02$@` is seen by Foundation as an object placeholder at position 2. (I didn't test those in a `stringdict` format key with variables names though, only with an hard-coded `printf`-like format in a playground)

Maybe we should just match `\d+` both here and in `PlaceholderTypes.swift` after all, and only filter out the position `0` after the fact, if found in the regex matches?

3. Also, I think it's a rare case where it's ok to force-unwrap the `NSRegularExpression(…)` here, because we're using a fixed and hard-coded pattern, so it should never fail once we ensured that the hard-coded regex we use works. Or we could just keep the `guard let` but `return []` (here and similarly for the ones you use for `Range(…, in:)` below). That would avoid the optional array as return value, and clean up some `guard let` at the call site upstream 🙂

Reply:
Just fixed all of those with recent commits:
 - Add a >10 placeholders entry and a test for it in 31125a572fde7499da18088f6cfc6b6839935a06. Test is coded with expected outcome so starts red
 - Fix the parsing logic for >10 position and make the test go green: https://github.com/SwiftGen/SwiftGen/pull/634/commits/e7590e47fc5585b62e48c706dda53b56e8aef3ec
 - Avoid optional return in 1a576133a8650bd29389d13f9fc640dc5f5d0d56